crypto/pkcs12/pkcs12-example.c

```diff
@@ -87,42 +101,40 @@ int main(int argc, char** argv)
     ret = wc_PKCS12_parse(pkcs12, "wolfSSL test", &keyDer, &keySz,
             &certDer, &certSz, &list);
     printf("return value of parsing pkcs12 = %d %s\n", ret, (ret == 0)? "SUCCESS": "FAIL");
-    if (ret != 0 || keyDer == NULL || certDer == NULL) {
+    if (ret != 0) {
         printf("\t error parsing pkcs12\n");
         wc_PKCS12_free(pkcs12);
         return -1;
     }
 
     /* print out key and cert found */
-    printf("HEX of Private Key Read (DER format) :\n");
-    for (i = 0; i < keySz; i++) {
-        if (i != 0 && !(i%16)) printf("\n");
-        printf("%02X", keyDer[i]);
-    }
-    printf("\n");
-
-    printf("\nHEX of Certificate Read (DER format) :\n");
-    for (i = 0; i < certSz; i++) {
-        if (i != 0 && !(i%16)) printf("\n");
-        printf("%02X", certDer[i]);
-    }
-    printf("\n");
-
     if (keyDer != NULL) {
+        printf("HEX of Private Key Read (DER format) :\n");
+        PRINT_BUFFER(keyDer, keySz);
         XFREE(keyDer, NULL, DYNAMIC_TYPE_PKCS);
     }
 
     if (certDer != NULL) {
+        printf("\nHEX of Certificate Read (DER format) :\n");
+        PRINT_BUFFER(certDer, certSz);
         XFREE(certDer, NULL, DYNAMIC_TYPE_PKCS);
     }
 
-    /* itterate through list if was not passed as null and free each node */
+    /* Iterate through list of certificates and print each out if was not passed
+     * as null, and then free each node. */
     if (list != NULL) {
         WC_DerCertList* current;
+        int certIdx = 0;
+
+        printf("\nHEX of Certificate LIST (DER format) :\n");
         current = list;
         while (current != NULL) {
-            WC_DerCertList* next = current->next;
+            WC_DerCertList* next;
+
+            next = current->next;
             if (current->buffer != NULL) {
+                printf("\n[CERT %d] :", certIdx++);
```

Comment:
Add newline after cert number: `[CERT %d] :\n`... Current output (without new-line) looks like this:
```
HEX of Certificate LIST (DER format) :

[CERT 0] :308204AA30820392A003020102020900
B7B69033661B6B23300D06092A864886
```
Adding new-line would align HEX output (looks a little cleaner!)